include/libsemigroups/sims.hpp

```diff
@@ -1668,35 +1626,65 @@ namespace libsemigroups {
 
     ~Sims2() = default;
 
+    //! \copydoc Sims1::Sims1(Presentation<Word> const&)
+    template <typename Word>
+    explicit Sims2(Presentation<Word> const& p) : Sims2() {
+      presentation(p);
+    }
+
+    //! \copydoc Sims1::Sims1(Presentation<Word> const&)
+    template <typename Word>
+    explicit Sims2(Presentation<Word> const&& p) : Sims2() {
+      presentation(std::move(p));
+    }
+
     //! \brief Reinitialize an existing Sims2 object.
     //!
-    //! This function puts a Sims2 object back into the same state as if
-    //! it had been newly default constructed.
+    //! This function puts an object back into the same state as if it had
+    //! been newly constructed from the presentation \p p.
     //!
-    //! \parameters (None)
+    //! \tparam Word the type of the words in the presentation \p p .
     //!
-    //! \returns A reference to \c this.
+    //! \param p the presentation.
     //!
-    //! \exception
-    //! \no_libsemigroups_except
-    Sims2& init() {
-      SimsSettings<Sims2>::init();
-      return *this;
-    }
-
-    //! \copydoc Sims1::Sims1(Presentation<Word> const&)
+    //! \returns A reference to \c *this.
+    //!
+    //! \throws LibsemigroupsException if `to_presentation<word_type>(p)` throws
+    //! \throws LibsemigroupsException if `p` is not valid
+    //! \throws LibsemigroupsException if `p` has 0-generators and 0-relations.
+    //!
+    //! \warning This function has no exception guarantee, the object will be
+    //! in the same state as if it was default constructed if an exception is
+    //! thrown.
+    //!
+    //! \sa presentation(Presentation<Word> const&)
     template <typename Word>
-    explicit Sims2(Presentation<Word> const& p) : Sims2() {
+    Sims2& init(Presentation<Word> const& p) {
+      init();
       presentation(p);
+      return *this;
     }
 
-    //! \copydoc Sims1::Sims1(Presentation<Word> const&)
+    //! \copydoc Sims2::init(Presentation<Word> const&)
     template <typename Word>
-    explicit Sims2(Presentation<Word> const&& p) : Sims2() {
+    Sims2& init(Presentation<Word> const&& p) {
+      init();
       presentation(std::move(p));
+      return *this;
     }
 
 #ifdef PARSED_BY_DOXYGEN
+    //! \brief Reinitialize an existing Sims2 object.
+    //!
+    //! This function puts a Sims2 object back into the same state as if
+    //! it had been newly default constructed.
+    //!
+    //! \returns A reference to \c this.
+    //!
+    //! \exception
```

Comment:
```suggestion
    //! \exceptions
```

See #628 for more info.